Bottom: 2c47feaacadc2816ecb0d75ffe61c5d9c6a2b7e8
Top:    1f4d3ed38b2ebddffbd20314579f927f6eb8598d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 01:20:26 +0100

Refresh of draw-c-split-gathering-from-re-0

---

diff --git a/src/draw.c b/src/draw.c
index 43015c5..08321b4 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -476,7 +476,7 @@ hole_counting_callback (const BoxType * b, void *cl)
 static int
 rat_callback (const BoxType * b, void *cl)
 {
-  DrawRat ((RatTypePtr) b);
+  draw_rat ((RatTypePtr) b);
   return 1;
 }
 
@@ -1335,18 +1335,8 @@ DrawLine (LayerTypePtr Layer, LineTypePtr Line)
 void
 DrawRat (RatTypePtr Line)
 {
-  if (!Gathering)
-    {
-      if (TEST_FLAG (SELECTEDFLAG | FOUNDFLAG, Line))
-	{
-	  if (TEST_FLAG (SELECTEDFLAG, Line))
-	    gui->set_color (Output.fgGC, PCB->RatSelectedColor);
-	  else
-	    gui->set_color (Output.fgGC, PCB->ConnectedColor);
-	}
-      else
-	gui->set_color (Output.fgGC, PCB->RatColor);
-    }
+  assert (Gathering);
+
   if (Settings.RatThickness < 20)
     Line->Thickness = pixel_slop * Settings.RatThickness;
   /* rats.c set VIAFLAG if this rat goes to a containing poly: draw a donut */
@@ -1354,28 +1344,47 @@ DrawRat (RatTypePtr Line)
     {
       int w = Line->Thickness;
 
-      if (Gathering)
-	{
-	  BoxType b;
+      BoxType b;
 
-	  b.X1 = Line->Point1.X - w * 2 - w / 2;
-	  b.X2 = Line->Point1.X + w * 2 + w / 2;
-	  b.Y1 = Line->Point1.Y - w * 2 - w / 2;
-	  b.Y2 = Line->Point1.Y + w * 2 + w / 2;
-	  AddPart(&b);
-	}
+      b.X1 = Line->Point1.X - w * 2 - w / 2;
+      b.X2 = Line->Point1.X + w * 2 + w / 2;
+      b.Y1 = Line->Point1.Y - w * 2 - w / 2;
+      b.Y2 = Line->Point1.Y + w * 2 + w / 2;
+      AddPart(&b);
+    }
+  else
+    DrawLineLowLevel ((LineType *) Line);
+}
+
+static void
+draw_rat (RatTypePtr line)
+{
+  if (TEST_FLAG (SELECTEDFLAG | FOUNDFLAG, line))
+    {
+      if (TEST_FLAG (SELECTEDFLAG, line))
+        gui->set_color (Output.fgGC, PCB->RatSelectedColor);
       else
-	{
-	  if (TEST_FLAG (THINDRAWFLAG, PCB))
-	    gui->set_line_width (Output.fgGC, 0);
-	  else
-	    gui->set_line_width (Output.fgGC, w);
-	  gui->draw_arc (Output.fgGC, Line->Point1.X, Line->Point1.Y,
-			 w * 2, w * 2, 0, 360);
-	}
+        gui->set_color (Output.fgGC, PCB->ConnectedColor);
+    }
+  else
+    gui->set_color (Output.fgGC, PCB->RatColor);
+
+  if (Settings.RatThickness < 20)
+    line->Thickness = pixel_slop * Settings.RatThickness;
+  /* rats.c set VIAFLAG if this rat goes to a containing poly: draw a donut */
+  if (TEST_FLAG(VIAFLAG, line))
+    {
+      int w = line->Thickness;
+
+      if (TEST_FLAG (THINDRAWFLAG, PCB))
+        gui->set_line_width (Output.fgGC, 0);
+      else
+        gui->set_line_width (Output.fgGC, w);
+      gui->draw_arc (Output.fgGC, line->Point1.X, line->Point1.Y,
+                     w * 2, w * 2, 0, 360);
     }
   else
-    _draw_line ((LineTypePtr) Line);
+    _draw_line ((LineType *) line);
 }
 
 /* ---------------------------------------------------------------------------
